src/ui/component/collapsible/collapsible.tsx

```diff
@@ -0,0 +1,52 @@
+import type { FC } from 'react'
+import { useState, useCallback } from 'react'
+import classNames from 'classnames'
+import * as RCollapsible from '@radix-ui/react-collapsible'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import type { AssertSubset } from '@/util/type'
+import './collapsible.scss'
+
+type CollapsibleProps = {
+  children: JSX.Element
+  triggerElement: JSX.Element
+  open?: boolean
+  triggerClassName?: string
+  chevronIconName?: AssertSubset<IconName, 'chevron' | 'chevron-sharp'>
+}
+
+export const Collapsible: FC<CollapsibleProps> = ({
+  open = false,
+  children,
+  triggerElement,
+  triggerClassName,
+  chevronIconName
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(open)
+
+  const handleOpenChange = useCallback((isOpen: boolean) => {
+    setIsOpen(isOpen)
+  }, [])
```

Comment:
why not:

```suggestion
  const handleOpenChange = setIsOpen
```

Reply:
Good suggestion! In this component we don't have any noticeable performance implications of not memoizing. It makes IMO the component easier to read. 

We can even directly use it in the JSX:
```tsx
<RCollapsible.Root onOpenChange={setIsOpen} open={isOpen}>
```